src/components/main-view/main-view.jsx

```diff
@@ -27,15 +33,25 @@ export class MainView extends React.Component {
 
   onMovieClick(movie) {this.setState({selectedMovie: movie }); }
   goBack() {this.setState({selectedMovie: null}); }
+  onLoggedIn(user) {this.setState({user}); }
+  startRegister() {this.setState({register: true}); }
+  endRegister() {this.setState({register: null}); }
 
   render() {
-    const { movies, selectedMovie } = this.state;
+    const { movies, selectedMovie, user, register } = this.state;
+
+    if (!register) {
+      return <RegistrationView endRegister={() => this.endRegister()}/>
+    }
+    
+    if (!user) return <LoginView onLoggedIn={user => this.onLoggedIn(user)}/>;
```

Comment:
Your login view has proptype warnings and missing props:
![Screenshot 2021-01-06 at 09 00 08](https://user-images.githubusercontent.com/30231189/103745335-92052e80-4fff-11eb-8585-def79ba7906d.png)

Reply:
After almost all day checking these issues - it seems i've no luck to find the mistakes for these 2 errors....

Reply:
This error occurs when you log in.
Errror nr. 1 says:  "startRegister" is marked as a required prop for the LoginView. But it is `undefined` when the LoginView is rendered. So if you render your LoginView how does the props object look like? Just log it. Or have a look at the React Dev Tools for Chrome. Then you will see there is no prop `startRegister`.  There is a prop called `onLoggedIn`. If you lool at the code above, why is the `onLoggedIn` prop defined and the `startRegister` isn't?
Error nr 2 happens on rendering the MovieCards. In PropTypes you set the genre title as required, but it is undefined at at least one of your movies.